src/Services/Subscriber/SubscriberUtil.php

```
@@ -67,31 +69,88 @@ public static function subscribe($name, $mobile, $group = false, $customFields =
             return esc_html__('To activate your subscription, the activation has been sent to your number.', 'wp-sms');
         } else {
+            SubscriberManager::forgetWelcomeMessageError();
+
             foreach ($groupIds as $groupId) {
                 // Add subscribe to database
                 $result = Newsletter::addSubscriber($name, $mobile, $groupId, '1', null, $customFields);
                 if ($result['result'] == 'error') {
+                    self::rollbackSubscribers($addedIds);
+
                     // Return response
                     return new \WP_Error('subscribe', $result['message']);
```

PHPStan (2.2.7)

[error] Method WP_SMS\Services\Subscriber\SubscriberUtil::subscribe() should return array|string but returns WP_Error.


(return.type)